drivers/i2c/i2c_ll_stm32_v1.c

```diff
@@ -229,17 +285,31 @@ s32_t stm32_i2c_msg_write(struct device *dev, struct i2c_msg *msg,
 	data->current.is_err = 0U;
 	data->slave_address = saddr;
 
+	if (LL_I2C_IsActiveFlag_BUSY(i2c)) {
```

Comment:
@ABOSTM busy is not supposed to be set at that point (and it will not go away on its own), what good would a timeout do? Or am I misunderstanding what you mean with "timeout" ?

Reply:
@lowlander , There is some short transitional state where Busy could be set:
Suppose you write a first byte. Driver will write to Data register.
From Hardware point of view, data is transferred to shadow register before being transferred on I2C bus. During this phase, and until STOP, the bus is busy. But it is possible to prepare for next Byte transfer: the TXE bit is set, meaning it is possible to write the next Byte to Data register.
(hardware is able to hold 1st byte in shadow register and next byte in Data register).
Thus from software point of view, after writting the 1st byte, all the conditions are met to be able to write the next one. So you may have hardware busy state while calling `stm32_i2c_msg_write()` and busy state will go away on its own once physical transfer is finished (after STOP generation).

Reply:
@ABOSTM ok good point, any suggestions for the timeout? one byte at 100kHz  ~ 100uS ?

Reply:
yes, but adding some margin... and timeout should never occurs so why not 1ms

Reply:
About timeout, there's a pending PR https://github.com/zephyrproject-rtos/zephyr/pull/21308
Can you check and see if it could be useful and/or superseded ?